### PR TITLE
Handle ENS Address Error

### PR DESCRIPTION
**Pull request by blackdevelopa**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**
When an ENS name does not have an address registered, we expect to return an [error](https://github.com/ethjs/ethjs-ens#enslookup-name-) to users.  However, on mobile [evan.eth](https://app.ens.domains/name/evan.eth/details) currently resolves to 0x0000.....0000. See reported [issue](https://github.com/MetaMask/metamask-mobile/issues/5396).

On mobile, [ethjs-ens](https://github.com/ethjs/ethjs-ens/issues/13) is used for resolving ens addresses. There's a known [issue](https://github.com/ethjs/ethjs-ens/issues/13) on said package. From the [code](https://github.com/ethjs/ethjs-ens/blob/8ea29591ae545a5da243b0f071b5676ff95aa647/index.js#L13), this PR returns when the `resolvedAddress` is `0x0000000000000000000000000000000000000000`

Test Case
_tested on mainnet_
* Tap to send eth
* Enter `evan.eth` on the receiver (to) input

**Screenshots/Recordings**
http://recordit.co/4c3dFLDytL

**Issue**

Progresses #5396

**Scenario**: Send to unregistered ENS address
Given I am attempting a send transaction
When I enter an ENS name not pointed to an eth address in the TO field
Then I am NOT able to complete a send transaction destined to 0x000...0000

**Scenario**: Send to 0's manually
Given I am attempting a send transaction
When I manually type/paste '0x0000000000000000000000000000000000000000' in the TO field
Then I am able to complete a send transaction destined to 0x000...0000

**Scenario**: Send to 0's saved as a contact
Given I have previously saved a contact with destination address '0x0000000000000000000000000000000000000000'
When I attempt a send transaction
And I select this previously saved contact as a destination in the TO field
Then I am able to complete a send transaction destined to 0x000...0000

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Review by jpuri — APPROVED** [review]

File: app/constants/transaction.ts [hunk 1]
```diff
@@ -15,3 +15,6 @@ export const TX_REJECTED = 'rejected';
 export const UINT256_BN_MAX_VALUE = new BN(2).pow(new BN(256)).sub(new BN(1));
 export const UINT256_HEX_MAX_VALUE =
   'ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff';
+
+// https://github.com/ethjs/ethjs-ens/blob/8ea29591ae545a5da243b0f071b5676ff95aa647/index.js#L13
+export const EMPTYADDRESS = '0x0000000000000000000000000000000000000000';
```
Comment (gantunesr) [inline app/constants/transaction.ts:20]: Nit
```suggestion
export const EMPTY_ADDRESS = '0x0000000000000000000000000000000000000000';
```

**Review by jpcloureiro — APPROVED** [review]

File: app/util/ENSUtils.js [hunk 2]
```diff
@@ -44,6 +45,7 @@ export async function doENSLookup(ensName, network) {
     this.ens = new ENS({ provider, network });
     try {
       const resolvedAddress = await this.ens.lookup(ensName);
+      if (resolvedAddress === EMPTYADDRESS) return;
```
[the remark this answers is not in the document]
Comment (jpcloureiro) [reply app/util/ENSUtils.js:48]: I think it is expected to persist the existing warning on the screen

see https://github.com/MetaMask/metamask-mobile/issues/5396#issuecomment-1400784449

File: app/constants/transaction.ts (hunk 1 above)
Comment (blackdevelopa) [reply app/constants/transaction.ts]: Thank you https://github.com/MetaMask/metamask-mobile/pull/5605/commits/85bb5df296bb2758e75880fd8659b05ebf6b7cce

**Review by plasmacorral — DISMISSED** [review]
This breaks deeplink payment requests to ENS addresses. This functions in production v5.13 build 1042 on my device (Samsung a515f w/Android11).

This can be tested by either tapping on a deeplink or scanning it as a QR code from wallet view.  

Video: https://recordit.co/ftn5q2HgMU

Expectation: Should be presented confirmation and have the chance to submit the transaction

Current State: Just returned to wallet view after engaging a payment deeplink to ENS address

Some example deeplinks:

https://metamask.app.link/send/0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48@1/transfer?address=evan.ethereum.eth&uint256=1e4

https://metamask.app.link/send/pay-metarnask.eth@1?value=1e13

**Comment by plasmacorral** [comment]
Ok...we made a production build to rule out any deeplinking issues in the debug build.  While payment deeplinks now work in build 1053 when a user clicks on them, they are still not functioning when scanned from the Qr scanner at wallet view.

Please test with these sample QR codes:
Screenshot 2023-01-26 at 1 01 00 PM
Screenshot 2023-01-26 at 1 00 47 PM

**Comment by plasmacorral** [comment]
Tested:
- Valid ENS deeplink Eth payment
- Valid ENS deeplink ERC20 payment
-  Invalid ENS deeplink
-  Intentionally sending to 0x000...000 to burn an asset
-  Payment deeplink on testnet without ENS
- Deeplink to DAPP

Note that scanning a QR to deeplink is broken on main currently, so this was not successfully tested.

**Pull request merged**